### ISSUE TITLE
Review hack description

### ISSUE THREAD

**Issue by ai** [ai/nanoid#165]
@MaxGraey @subzey @CyberAP @reklatsmasters @oriSomething I added description for all your amazing hacks. Can you double-check these descriptions?

https://github.com/ai/nanoid/blob/master/random.js
https://github.com/ai/nanoid/blob/master/index.browser.js
https://github.com/ai/nanoid/blob/master/format.js
https://github.com/ai/nanoid/blob/master/format.browser.js
https://github.com/ai/nanoid/blob/master/non-secure/generate.js
https://github.com/ai/nanoid/blob/master/non-secure/index.js
https://github.com/ai/nanoid/blob/master/async/format.browser.js
https://github.com/ai/nanoid/blob/master/async/format.js
https://github.com/ai/nanoid/blob/master/async/index.browser.js
https://github.com/ai/nanoid/blob/master/async/index.js
https://github.com/ai/nanoid/blob/master/async/random.js

**Comment by ai** [comment]
@oriSomething thanks, fixed https://github.com/ai/nanoid/commit/546747ee9ae90a2d57740adc552b95727653ab44

I decided to avoid details about BigNum, since we didn’t document side effects for other hacks

**Comment by reklatsmasters** [comment]
https://github.com/ai/nanoid/blob/1e590b92f0f71457e9d42d49f792457598860f2f/format.js#L32
Should be `x^2`?

LGTM for description. :+1: 

---

My small note about async model in nanoid. What is **true async call**? When a hard job doing in background task, like here:

```js
const randomBytes = promisify(crypto.randomBytes);
await randomBytes(10);
```

Here is cryptography-strong random generator from OpenSSL called  in background task. What is **false async call**? It's sync calls disguised as async one:

```js
Promise.resolve(crypto.getRandomValues(new Uint8Array(bytes)))
```
Here is delayed value. All crypto primitives from `getRandomValues` was called in main thread.

**Comment by ai** [comment]
> Should be x^2?

For 8 symbols alphabet we need `00000111` mask. `111 base 2` = `7 base 10`. `7` === `2 ** 3 - 1`.

Seems like we need to fix `x^^2 - 1` to `2 ^^ x - 1`. I fixed it 532e275

> My small note about async model in nanoid
